### PR TITLE
Update Electrum to 4.0.9

### PR DESCRIPTION
**Pull request by ar-jan**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

#### Does it build and run successfully? 
(Please choose at least one native build and, if supported, at least one cross build. More are better.)
- [x] I built this PR locally for my native architecture, (ARCH-LIBC)
- [ ] I built this PR locally for these architectures (if supported. mark crossbuilds):
  - [ ] aarch64-musl
  - [ ] armv7l
  - [ ] armv6l-musl

~This depends on #28272.~

~Using package [libbitcoin-secp256k1](https://github.com/void-linux/void-packages/tree/master/srcpkgs/libbitcoin-secp256k1) as dependency might be possible, but: it is based on this release: https://github.com/libbitcoin/secp256k1/releases/tag/v0.1.0.13 with the last commit included dating to December 2016. Electrum currently [uses](https://github.com/spesmilo/electrum/blob/master/contrib/make_libsecp256k1.sh) a commit from from June 2020 from https://github.com/bitcoin-core/secp256k1.~

--> The old version of `libbitcoin-secp256k1` in Void's repos is is sufficient for now, as per [this Electrum issue](https://github.com/spesmilo/electrum/issues/7014): `if the tests pass, it is considered good enough.`

New dependencies:

    python3-socks-1.2.0
    python3-curio-1.4
    python3-trio-0.18.0
    python3-outcome-1.1.0
    python3-sniffio-1.2.0
    python3-bitstring-3.1.7

Updated dependencies:

    python3-aiohttp_socks: update to 0.5.5.
    python3-dnspython: update to 2.1.0. 

So this is ready; also tested locally.

### CODE REVIEW

**Comment by ar-jan** [comment]
Hello @clehner, I've updated Electrum to v 4.0.9 but I'm not sure how best to proceed with providing `libsecp256k1`. Do you have a suggestion?

**Comment by ar-jan** [comment]
Hi @atweiden, @CameronNemo, I was wondering if either of you has a suggestion for how to provide an update to `libsecp256k1` for Void (as maintainer and user of the library respectively). ElectronCash uses the package that's currently in Void, which is based on a very old version of `libbitcoin/secp256k1`.
Electrum 4.x uses a more recent version based on `bitcoin-core/secp256k1`, which is also the upstream repo for libbitcoin/secp256k1. I've updated Electrum but I still need a way to provide a more recent version of the library.

**Comment by atweiden** [comment]
Looks like secp256k1 upstream still hasn't tagged a release. If the Electrum maintainers want you to use upstream secp256k1, you probably shouldn't use the one from libbitcoin.

**Comment by ar-jan** [comment]
I [asked](https://github.com/spesmilo/electrum/issues/7014) and it looks like the libbitcoin version may work, for now at least. But I'll  need to run Electrum tests and that'll require packaging a few more dependencies first.

**Comment by ar-jan** [comment]
The move back to OpenSSL allowed Electrum tests to pass. This version uses the very old `libbitcoin-secp256k1` package, but is is sufficient for now, as per [this Electrum issue](https://github.com/spesmilo/electrum/issues/7014):

> if the tests pass, it is considered good enough. [to use the old libbitcoin-secp256k1]

I think this is ready now.

**Comment by ericonr** [comment]
Try adding `iana-etc` to `checkdepends`.

**Comment by ar-jan** [comment]
@ericonr Yes thanks, this has solved the issue for the arches that were failing, but now `x86_64-musl` is failing?

**Comment by ericonr** [comment]
Alpine is using 2.1.0 of dnspython and doesn't seem to disable any tests.

File: srcpkgs/electrum/template [hunk 1]
```diff
@@ -1,25 +1,31 @@
...
+# Optional dependencies:
 #  btchip - BTChip hardware wallet support
 #  trezor - TREZOR hardware wallet support
 #  keepkey - KeepKey hardware wallet support
 #  qdarkstyle - Dark theme
+checkdepends="python3-pytest python3-attrs python3-aiohttp>=3.3.0<4.0.0
+ python3-aiohttp_socks>=0.3 python3-aiorpcx>=0.18<0.19 python3-bitstring
+ python3-dnspython python3-ecdsa python3-jsonrpclib python3-protobuf
+ python3-pyaes python3-pycryptodomex python3-socks python3-qrcode
+ python3-PyQt5 python3-QtPy python3-cryptography libbitcoin-secp256k1"
```
Comment (ericonr) [inline srcpkgs/electrum/template:21]: It's easier to do `checkdepends="$depends python3-pytest ..."`

File: srcpkgs/python3-trio/template [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+distfiles="${PYPI_SITE}/t/trio/trio-${version}.tar.gz"
+checksum=87a66ae61f27fe500c9024926a9ba482c07e1e0f56380b70a264d19c435ba076
+
+# Tests also need the following to be packaged:
+# pyOpenSSL https://pypi.org/project/pyOpenSSL/
+# trustme https://pypi.org/project/trustme/
+# astor https://pypi.org/project/astor/
+do_check() {
+	:
+}
```
Comment (ericonr) [inline srcpkgs/python3-trio/template:27]: Use `make_check=no` instead of defining the function.

File: srcpkgs/python3-trio/template [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+version=0.18.0
+revision=1
+wrksrc="trio-${version}"
+build_style=python3-module
+hostmakedepends="python3-setuptools"
+depends="python3-async_generator>=1.9 python3-attrs>=19.2.0 python3-idna
+ python3-outcome python3-sniffio python3-sortedcontainers"
+checkdepends="python3-pytest python3-pytest-cov python3-ipython python3-pylint
+ python3-jedi black python3-mypy python3-flake8
+ python3-async_generator>=1.9 python3-attrs>=19.2.0
```
Comment (ericonr) [inline srcpkgs/python3-trio/template:12]: If the restriction is just a minimum version, it isn't really necessary.

File: srcpkgs/python3-dnspython/template [hunk 4]
```diff
@@ -1,17 +1,26 @@
...
-homepage="http://www.dnspython.org/"
+homepage="https://www.dnspython.org/"
 distfiles="${PYPI_SITE}/d/dnspython/dnspython-${version}.zip"
-checksum=36c5e8e38d4369a08b6780b7f27d790a292b2b08eea01607865bf0936c558e01
+checksum=e4a87f0b573201a0f3727fa18a516b055fd1107e0e5477cded4a2de497df1dd4
+
+do_check() {
+	# This exclusion can be removed for 2.1.1 or 2.2.0 (https://github.com/rthalley/dnspython/issues/622)
+	PYTHONPATH=$(cd build/lib* && pwd) pytest -k 'not test_unpickle'
+}
```
Comment (ericonr) [inline srcpkgs/python3-dnspython/template:23]:
```suggestion
checksum=e4a87f0b573201a0f3727fa18a516b055fd1107e0e5477cded4a2de497df1dd4
# tests not working on CI
make_check=extended

do_check() {
	# This exclusion can be removed for 2.1.1 or 2.2.0 (https://github.com/rthalley/dnspython/issues/622)
	PYTHONPATH=$(cd build/lib* && pwd) pytest -k 'not test_unpickle'
}
```

**Review by ericonr — COMMENTED** [review]
That's a lot of new packages >.<

File: srcpkgs/python3-dnspython/template (hunk 4 above)
Comment (ericonr) [reply srcpkgs/python3-dnspython/template:23]:
```
=================================== FAILURES ===================================
____________ OverrideSystemResolverTestCase.test_basic_getaddrinfo _____________

self = <tests.test_resolver_override.OverrideSystemResolverTestCase testMethod=test_basic_getaddrinfo>

    @unittest.skipIf(sys.platform == 'win32',
                     'avoid windows original getaddrinfo issues')
    def test_basic_getaddrinfo(self):
>       self.assertTrue(self.equivalent('dns.google', 53, socket.AF_INET,
                                        socket.SOCK_DGRAM))
E       AssertionError: False is not true

tests/test_resolver_override.py:87: AssertionError
----------------------------- Captured stdout call -----------------------------
NOT EQUIVALENT
[(<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_DGRAM: 2>, 17, '', ('8.8.8.8', 53)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_DGRAM: 2>, 17, '', ('8.8.4.4', 53))]
[(<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_DGRAM: 2>, 17, 'dns.google', ('8.8.4.4', 53)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_DGRAM: 2>, 17, 'dns.google', ('8.8.8.8', 53))]
=========================== short test summary info ============================
FAILED tests/test_resolver_override.py::OverrideSystemResolverTestCase::test_basic_getaddrinfo
========== 1 failed, 1149 passed, 25 skipped, 1 deselected in 10.25s ===========
```

kinda still broken on musl, seems to be the same errors >.<

**Comment by ar-jan** [comment]
Thanks @ericonr. I read the description on `make_check=extended` but am not sure why / what exactly the difference is? (but included it now).

> That's a lot of new packages >.<

Indeed! I just wanted to update this one package, not sure if I would've started with this had I known :S

It seems the problems on musl were intermittent, since they passed before and do so again now.

**Comment by ar-jan** [comment]
Hello @Chocimier, care to take a look at this, since you reviewed these dependencies earlier in #28272? I think this is ready.

File: srcpkgs/electrum/template [hunk 5]
```diff
@@ -1,25 +1,27 @@
...
 build_style=python3-module
-pycompile_module="electrum electrum_gui electrum_plugins"
-hostmakedepends="python3-PyQt5-devel-tools python3-setuptools"
-depends="python3-aiohttp python3-aiohttp_socks python3-aiorpcx>=0.18<0.19
- python3-dnspython python3-ecdsa python3-jsonrpclib python3-protobuf
- python3-pyaes python3-pycryptodome python3-PyQt5 python3-qrcode"
-# Missing optional dependencies:
+hostmakedepends="python3-setuptools python3-PyQt5-devel-tools"
+depends="python3-aiohttp>=3.3.0<4.0.0 python3-aiohttp_socks>=0.3
+ python3-aiorpcx>=0.18<0.19 python3-bitstring python3-dnspython python3-ecdsa
```
Comment (ericonr) [inline srcpkgs/electrum/template:8]: I don't like the version restrictions here, and would prefer to leave them unspecified.

File: srcpkgs/electrum/patches/no-certifi.patch [hunk 6]
```diff
@@ -2,28 +2,31 @@ Upstream: No
 Reason: Avoid certifi dependency
```
Comment (ericonr) [inline srcpkgs/electrum/patches/no-certifi.patch:2]: Does this actually help? I'd rather add the dependency than carry this forever...

Comment (ar-jan) [reply srcpkgs/electrum/patches/no-certifi.patch]: Indeed, no reason for this now that it's available in the repository.

File: srcpkgs/electrum/template (hunk 5 above)
Comment (ar-jan) [reply srcpkgs/electrum/template]: Done!

**Pull request merged**